### PR TITLE
Add location/position support

### PR DESCRIPTION
**Pull request by globophobe**
Should fix issue #11. Tests pass. 

It's my first time using ctypes. Hopefully, there aren't any mistakes.

### CODE REVIEW

File: pyzbar/wrapper.py [hunk 1]
```diff
@@ -261,6 +261,26 @@ def zbar_function(fname, restype, *args):
...
 
+zbar_symbol_get_loc_size = zbar_function(
+    'zbar_symbol_get_loc_size',
+    c_uint,
+    POINTER(c_int)         # values in ZBARSymbol
+)
+
+zbar_symbol_get_loc_x = zbar_function(
+    'zbar_symbol_get_loc_x',
+    c_uint,
```
Comment (quicklizard99) [inline pyzbar/wrapper.py]: My copy of `zbar.h` has the return type as `int`, which is represented by `c_int`. Could you check the type in your copy of the header?

File: pyzbar/wrapper.py [hunk 2]
```diff
@@ -261,6 +261,26 @@ def zbar_function(fname, restype, *args):
...
+    POINTER(c_int),        # values in ZBARSymbol
+    c_uint
+)
+
+zbar_symbol_get_loc_y = zbar_function(
+    'zbar_symbol_get_loc_y',
+    c_uint,
```
Comment (quicklizard99) [inline pyzbar/wrapper.py]: My copy of `zbar.h` has the return type as `int`, which is represented by `c_int`. Could you check the type in your copy of the header?

File: pyzbar/pyzbar.py [hunk 3]
```diff
@@ -11,14 +11,15 @@
     zbar_image_create, zbar_image_destroy, zbar_image_set_format,
     zbar_image_set_size, zbar_image_set_data, zbar_scan_image,
     zbar_image_first_symbol, zbar_symbol_get_data,
+    zbar_symbol_get_loc_size, zbar_symbol_get_loc_x, zbar_symbol_get_loc_y,
     zbar_symbol_next, ZBarConfig, ZBarSymbol, EXTERNAL_DEPENDENCIES
 )
 
 __all__ = ['decode', 'EXTERNAL_DEPENDENCIES']
```
Comment (quicklizard99) [inline pyzbar/pyzbar.py]: Suggest that we return a single rectangle that describes the bounding box around the detected barcode:

```
# A rectangle
Rect = namedtuple('Rect', ['left', 'top', 'width', 'height'])

 # Results of reading a barcode
Decoded = namedtuple('Decoded', ['data', 'type', 'rect'])
```

File: pyzbar/pyzbar.py [hunk 4]
```diff
@@ -144,9 +145,22 @@ def decode(image, symbols=None):
...
+
+                    location = []
+                    loc = zbar_symbol_get_loc_size(symbol)
+                    if loc:
+                        location = [
+                            (
+                                zbar_symbol_get_loc_x(symbol, l),
+                                zbar_symbol_get_loc_y(symbol, l)
+                            )
+                            for l in range(loc)
```
Comment (quicklizard99) [inline pyzbar/pyzbar.py]: Prefer `xrange` on Python 2.x . At the top of the module (or maybe within this function):

```
RANGEFN = getattr(globals(), 'xrange', range)
```

Alter `range(loc)` to `RANGEFN(loc)`

File: pyzbar/pyzbar.py [hunk 5]
```diff
@@ -144,9 +145,22 @@ def decode(image, symbols=None):
...
+                        ]
+
                     results.append(Decoded(
                         data=data,
-                        type=symbol_type
+                        type=symbol_type,
+                        location=location
```
Comment (quicklizard99) [inline pyzbar/pyzbar.py]: Replace with `rect=bounding_box_of_locations(locations)`

File: pyzbar/pyzbar.py [hunk 6]
```diff
@@ -11,14 +11,15 @@
...
+    zbar_symbol_get_loc_size, zbar_symbol_get_loc_x, zbar_symbol_get_loc_y,
     zbar_symbol_next, ZBarConfig, ZBarSymbol, EXTERNAL_DEPENDENCIES
 )
 
 __all__ = ['decode', 'EXTERNAL_DEPENDENCIES']
 
 
 # Results of reading a barcode
-Decoded = namedtuple('Decoded', ['data', 'type'])
+Decoded = namedtuple('Decoded', ['data', 'type', 'location'])
```
Comment (quicklizard99) [inline pyzbar/pyzbar.py]: A function to compute bounding boxes from scan locations (uses `operator.itemgetter`):

```
def bounding_box_of_locations(locations):
    x_values = list(map(itemgetter(0), locations))
    x_min, x_max = min(x_values), max(x_values)
    y_values = list(map(itemgetter(1), locations))
    y_min, y_max = min(y_values), max(y_values)
    return (Rect(x_min, y_min, x_max - x_min,  y_max - y_min))
```

File: pyzbar/pyzbar.py [hunk 7]
```diff
@@ -144,9 +145,22 @@ def decode(image, symbols=None):
                 while symbol:
                     data = string_at(zbar_symbol_get_data(symbol))
                     symbol_type = ZBarSymbol(symbol.contents.value).name
+
+                    location = []
+                    loc = zbar_symbol_get_loc_size(symbol)
+                    if loc:
```
Comment (quicklizard99) [inline pyzbar/pyzbar.py]: I am curious - does `zbar_symbol_get_loc_size` ever return `0`? I have done some limited testing on Mac OS X and have always seen a non-zero return.

**Comment by globophobe** [comment]
Updated the pull request. Thank you for your suggestions.

**Comment by quicklizard99** [comment]
Fantastic! Thanks for the very fast update @globophobe

**Pull request merged**